### ISSUE TITLE
Compiler asserts if static methodmap function is used before declaration

### ISSUE THREAD

**Issue by Kailo97** [alliedmodders/sourcepawn#196]
Friend catched this case while coding plugin. Here my example for demonstrating it.
crash1.sp
```
#include "crash2.sp"
public void OnPluginStart()
{
	DoIt(); // - Shia LaBeouf
}
```
crash2.sp
```
#include "crash3.sp"
methodmap Shia
{
	public static Speak()
	{
		PrintToServer("Do it! - Shia LaBeouf");
	}
};
```
crash3.sp
```
void DoIt()
{
	Shia.Speak();
}
```
log
```
crash3.sp(3) : error 017: undefined symbol "Shia"
```
Compiler crashed while compilation.
If move `#include "crash3.sp"` below mathodmap - everything ok.
Test environment: Windows 10, SM 1.8.0.6041 & 1.9.0.6226

**Comment by Fyren** [comment]
```
public void main() { mm.sFunc(); }

methodmap mm { public static void sFunc() {} };
```

If you try to declare a variable and call a non-static instead, you get the expected (but perhaps undesired) error about it not being declared.

**Comment by Kailo97** [comment]
Has this been fixed in #256?

**Comment by Fyren** [comment]
My given test case no longer asserts and triggers the error added in that PR.  I don't know if there was anything further that needed changing, but since it's been a year, I assume no one else remembers, either.
